src/protocol.h

```diff
@@ -373,16 +373,7 @@ class CAddress : public CService
         if (s.GetType() & SER_DISK) {
             READWRITE(nVersion);
         }
-        if ((s.GetType() & SER_DISK) ||
-            (nVersion != INIT_PROTO_VERSION && !(s.GetType() & SER_GETHASH))) {
```

Comment:
By removing this we assume that for non-disk serialization `s.GetType() & SER_GETHASH` is always false. I.e. `SER_GETHASH` is not used in non-disk ser/deser of `CAddress`. I think this is the case indeed.

Reply:
Yes, this is the case. #20516 adds an assert for this, actually.